src/main/webapp/components/workspace/workspace.js

```diff
@@ -217,6 +220,10 @@ export default () => {
                 }, [])
                 setListResults(results)
               }}
+              onSave={lists => {
```

Comment:
I think it would be better if you only saved the lists currently stored in `listState`, and pass down another callback to change list state. You are currently changing the list state and also saving a version not in sync with the one stored on the workspaces component. I think that separating these concerns and keeping a single source of truth will make things easier to reason about as more functionality is added.